### ISSUE TITLE
False positive in javascript.lang.correctness.useless-assign.useless-assignment

### ISSUE THREAD

**Issue by Sjord** [returntocorp/sgrep-rules#3036]
[semgrep-rules/javascript/lang/correctness/useless-assign.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/javascript/lang/correctness/useless-assign.yaml)

test.js:

```
const { foo } = require('foo')
const { bar } = require('bar')
```

```
$ semgrep -c 'r/javascript.lang.correctness.useless-assign.useless-assignment' test.js
...
┌────────────────┐
│ 1 Code Finding │
└────────────────┘

    test.js
       javascript.lang.correctness.useless-assign.useless-assignment
          `const` is assigned twice; the first assignment is useless
          Details: https://sg.run/0Q9w

            2┆ const { foo } = require('foo')
            3┆ const { bar } = require('bar')
```

**Comment by nashcontrol** [comment]
I thought it was related to the regex implementation - https://github.com/returntocorp/semgrep/issues/5216#issuecomment-1651714473

maybe this one is duplicate of https://github.com/returntocorp/semgrep/issues/7531
